pkg/scheduler/allocator.go

```diff
@@ -214,6 +211,56 @@ func (m *Scheduler) tryBatchAllocation(partition string, candidates []*Schedulin
     return allocations, failedAsks
 }
 
+// TODO: convert this as an interface.
```

Comment:
Yeah make sense, this should be part of node_sorting_policy, it should return an iterator of sorted node list.